policies/documentation-policy.md

```diff
@@ -0,0 +1,91 @@
+documentation is in _POD_ format.
+
+## Chapter 2: Public symbols in the libraries
+
+All new public symbols must be documented in a _POD_ manual page in the
+`doc/man3` directory.  This includes both macros and functions.
+
+The allowed exceptions are:
+
+- guard macros preventing a header file being included twice
+- new OPENSSL_NO_DEPRECATED_ macros
+- new symbols generated automatically via `make update` (errors, objects, etc)
+
+## Chapter 3: Internal functions, structures, globals and macros
+
+Internal functions and macros are those defined in the `include/crypto`
+and `include/internal` directories.
+
+These should all be documented at the point of definition.  A comment
+describing their purpose and the input and output arguments for functions
+suffices.
+
+For _trivial_ items, where their operation is obvious from their
+definition, the documentation requirement is not required.  The following
+are generally representative of trivial items, however it is quite
+possible for any of these to be non-trivial in specific instances and
+therefore require documentation:
+
+- OSSL_DISPATCH tables
+- method structures
+- upref functions
+- free functions
+- simple getter/setter functions
+- wrappers for other functions (a function that calls a more recent `_ex`
+  variant)
+
+For structures, each of the fields should be commented stating its
+purpose.  Again, a _trivial_ exception applies where the purpose is
+obvious.  Some representative examples:
+
+- `OSSL_LIB_CTX *ctx;` where there is only one library context referenced in
+  the structure.
+- `struct *next;` in a linked list implementation.
+- `CRYPTO_REF_COUNT refcnt;`
+- `CRYPTO_RWLOCK *lock;` where there is only one lock in the structure.
+
+## Chapter 4: Static functions and globals and local structures and macros
+
+These are functions, structures, globals and macros local to a specific
+C file or defined for a single directory as part of a local .h file.
+
+These should all be documented at the point of definition.  Follow the
+rules and exceptions listed in Chapter 3.  In some cases slightly more
+leniency with respect to _trivial_ can be tolerated.
+
+## Chapter 5: Code comments
+
+Comments in code should explain what the algorithm is doing, not each
+statement.  As the complexity of the code increases, the size and detail
+of comments should also increase.
+
+## Chapter 6: Assembly code
+
+Assembly code should include a good description of the algorithm and
+approach being used.  This should be followed by a performance comparison
+and then the assembly code itself.  The assembly code should be well
+commented, but it isn't necessary to comment every line.  A comment
+describing each block of code suffices.
+
+There are no _trivial_ exceptions for assembly code.
+
+## Chapter 7: Automated sanity checking
+
+The `make doc-nits` command should be run before submitting a pull
+request and any problems it locates must be addressed.
+
```

Comment:
This policy doesn't seem to cover:

- New Configure options (i.e. INSTALL.md)
- CHANGES.md and NEWS.md

Reply:
I meant to but forgot about it.  I'll add sections for these.